### ISSUE TITLE
Implementation of fetch doesn't work in browser?

### ISSUE THREAD

**Issue by sdbondi** [feathersjs/feathers-rest#62]
I get the following error when using fetch in the browser (polyfilled, but using native window.fetch) when the call returns a http 400:

`body.on is not a function` over here: https://github.com/feathersjs/feathers-rest/blob/master/src/client/fetch.js#L31

Shouldn't `response.json()` be used to parse the body even in the case of an error?
[Body](https://developer.mozilla.org/en-US/docs/Web/API/Body) has no `on` method.

**Comment by jomcode** [comment]
Also experiencing this issue in a project. In chrome dev tools under the network tab if I look at the response I get something like the following: 
```
{"name":"NotAuthenticated","message":"Invalid login.","code":401,"className":"not-authenticated","errors":{}}
```

But in the catch block the error is the same `body.on is not a function` that is posted above.

I'm currently using core-js es6/7 and whatwg-fetch polyfills.

I configure the feathers client like the following:
```
const app = feathers()
  .configure(rest(rootUrl).fetch(window.fetch.bind(window)))
```

**Comment by daffl** [comment]
Fixed in 1.4.3, new `feathers-client` release shortly.

**Comment by sdbondi** [comment]
@daffl Wow, thanks for the quick response!
